Automattic/sensei
# Rename course list to item list and add filters

Fixes #

### Changes proposed in this Pull Request

* Rename CourseList to ItemList.
* Add filters to add more flexibility and enable adding custom bulk actions.

### Testing instructions

* Checkout `add/student-bulk-actions-filters`.
* `npm install`, `npm run start`.
* Go to Sensei LMS -> Students, try different bulk actions.
* Bulk actions must work as expected.

### New/Updated Hooks

* `senseiStudentModalPossibleActions` – Filters possible actions in the Student Modal.
* `senseiStudentModalItemListEmpty` – Filters the message for the empty list in ItemList.
* `senseiStudentModalItemListQuery` – Filters the query to get the item list for Student Modal.
* `senseiStudentModalItemListPostType` – Filters the post type of the items in the list in the Student Modal.
* `senseiStudentModalItemListTitle` – Filters the title for the item list in the Student Modal.
* `senseiStudentModalTitle` – Filters Student Modal title.
* `senseiStudentModalSearchItemsPlaceholder` – Filters possible actions in the Student Modal.
